pages/index.js

```diff
@@ -32,26 +36,25 @@ const Home = ({ doc, posts }) => {
   return <SetupRepo />;
 };
 
-export async function getStaticProps({ preview = null, previewData = {} }) {
-
-  const { ref } = previewData
+export async function getStaticProps({ previewData }) {
 
-  const client = Client()
+  const ref = previewData ? previewData.ref : null
+  const options = ref ? { ref } : null
 
-  const doc = await client.getSingle("blog_home", ref ? { ref } : null) || {}
+  const doc = await Client().getSingle("blog_home", options) || {}
 
-  const posts = await client.query(
+  const posts = await Client().query(
     Prismic.Predicates.at("document.type", "post"), {
       orderings: "[my.post.date desc]",
-      ...(ref ? { ref } : null)
+      ...(options)

```

Comment:
Also move the definition of the query options here to its own line.

Reply:
So something like this?

```
  const previewRef = previewData ? previewData.ref : null
  const options = previewRef ? { ref: previewRef } : null

  const doc = await Client().getSingle("blog_home", options) || {}

  const queryOptions = { orderings: "[my.post.date desc]", ...(options)},

  const posts = await Client().query(Prismic.Predicates.at("document.type", "post"), queryOptions)
```

Reply:
@phillysnow Yeah this is more what I was thinking. I would rename some of the variables, though.

```
  const previewRef = previewData ? previewData.ref : null
  const refOption = previewRef ? { ref: previewRef } : null

  const blogHome = await Client().getSingle('blog_home', refOption) || {}

  const postQueryOptions = { orderings: '[my.post.date desc]', ...refOption },
  const posts = await Client().query(Prismic.Predicates.at('document.type', 'post'), postQueryOptions)
```